server/src/main/java/org/elasticsearch/script/DynamicMap.java

```diff
@@ -98,6 +98,17 @@ public Set<Entry<String, Object>> entrySet() {
 
     @Override
     public String toString() {
-        return delegate.toString();
+        StringBuilder sb = new StringBuilder("DynamicMap [");
+        int count = 0;
+        for (Map.Entry<String, Object> entry : entrySet()) {
+            sb.append(entry.getKey());
+            sb.append(": ");
```

Comment:
We should the same format as other Map's (eg HashMap, Map.of, etc) in Java use, which I believe is with curly braces and `=` between key/values. Also I wonder if we need the `DynamicMap` prefix?